src/Stratis.Sidechains.Networks/FederatedPegMain.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using NBitcoin;
+using NBitcoin.DataEncoders;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.Features.PoA;
+
+namespace Stratis.Sidechains.Networks
+{
+    /// <summary>
+    /// Right now, ripped nearly straight from <see cref="PoANetwork"/>.
+    /// </summary>
+    public class FederatedPegMain : PoANetwork
+    {
+        public Key[] FederationKeys { get; private set; }
+
+        internal FederatedPegMain()
+        {
+            this.Name = FederatedPegNetwork.MainNetworkName;
+            this.CoinTicker = FederatedPegNetwork.CoinSymbol;
+
+            var consensusFactory = new PoAConsensusFactory();
+
+            // Create the genesis block.
+            this.GenesisTime = 1513622125;
+            this.GenesisNonce = 1560058197;
+            this.GenesisBits = 402691653;
+            this.GenesisVersion = 1;
+            this.GenesisReward = Money.Zero;
+
+            NBitcoin.Block genesisBlock = CreatePoAGenesisBlock(consensusFactory, this.GenesisTime, this.GenesisNonce, this.GenesisBits, this.GenesisVersion, this.GenesisReward);
+            //((SmartContractPoABlockHeader)genesisBlock.Header).HashStateRoot = new uint256("21B463E3B52F6201C0AD6C991BE0485B6EF8C092E64583FFA655CC1B171FE856"); // Set StateRoot to empty trie.
+
+            this.Genesis = genesisBlock;
+
+            // Keeping the 3rd there in case we use it in future. For our integration tests we use 2 nodes currently.
+            this.FederationKeys = new Key[]
+            {
+                new Mnemonic("lava frown leave wedding virtual ghost sibling able mammal liar govern wisdom").DeriveExtKey().PrivateKey,
+                new Mnemonic("idle power swim wash diesel blouse photo among eager reward wide hair").DeriveExtKey().PrivateKey,
+                new Mnemonic("high neither night category fly wasp inner kitchen phone current skate menu").DeriveExtKey().PrivateKey
+            };
+
+            var federationPubKeys = new List<PubKey>
+            {
+                this.FederationKeys[0].PubKey,
+                this.FederationKeys[1].PubKey,
+                this.FederationKeys[2].PubKey
+            };
+
+            var consensusOptions = new PoAConsensusOptions(
+                maxBlockBaseSize: 1_000_000,
+                maxStandardVersion: 2,
+                maxStandardTxWeight: 100_000,
+                maxBlockSigopsCost: 20_000,
+                maxStandardTxSigopsCost: 20_000 / 5,
+                federationPublicKeys: federationPubKeys,
+                targetSpacingSeconds: 3
```

Comment:
this needs to be 16. it's the time between block production. anything less than about 13 is uncharted territory.